2024-09-26-typescript-effect/src/index.ts

```diff
@@ -0,0 +1,195 @@
+import { Effect, Array, Option, pipe, Stream, Chunk, StreamEmit } from "effect";
+import { Schema } from "@effect/schema";
+import {
+  FetchHttpClient,
+  HttpClient,
+  HttpClientResponse,
+} from "@effect/platform";
+
+const CityResponse = Schema.Struct({
+  name: Schema.String,
+  country_code: Schema.String.pipe(Schema.length(2)),
+  latitude: Schema.Number,
+  longitude: Schema.Number,
+});
+
+type CityResult = Schema.Schema.Type<typeof CityResponse>;
+
+const GeocodingResponse = Schema.Struct({
+  results: Schema.Array(CityResponse),
+});
+
+type GeocodingResponse = Schema.Schema.Type<typeof GeocodingResponse>;
+
+const Weather = Schema.Struct({
+  current_units: Schema.Struct({
+    temperature_2m: Schema.String,
+    relative_humidity_2m: Schema.String,
+    apparent_temperature: Schema.String,
+    precipitation: Schema.String,
+  }),
+  current: Schema.Struct({
+    temperature_2m: Schema.Number,
+    relative_humidity_2m: Schema.Number,
+    apparent_temperature: Schema.Number,
+    precipitation: Schema.Number,
+  }),
+});
+
+type Weather = Schema.Schema.Type<typeof Weather>;
+
+// The field input
+const city = Option.fromNullable(
+  document.querySelector<HTMLInputElement>("#city"),
+);
+// The list of suggestions
+const cities = Option.fromNullable(
+  document.querySelector<HTMLUListElement>("#cities"),
+);
+// The weather information
+const weather = Option.fromNullable(
+  document.querySelector<HTMLDivElement>("#weather"),
+);
+
+const getRequest = (url: string) =>
```

Comment:
Can you pls add some return types to all the functions? Imho it would help when reading the code to understand them better.

Reply:
Done